frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx

```diff
@@ -0,0 +1,257 @@
+    undefined,
+  );
+
+  const { updateLeaseDeposit, removeLeaseDeposit } = useLeaseDeposits();
+  const { updateLeaseDepositReturn, removeLeaseDepositReturn } = useLeaseDepositReturns();
+
+  const onAddDeposit = () => {
+    setEditDepositValue(FormLeaseDeposit.createEmpty());
+    setShowEditModal(true);
+  };
+
+  const onEditDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditDepositValue(FormLeaseDeposit.createFromModel(deposit));
+      setShowEditModal(true);
+    }
+  };
+
+  const onDeleteDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setDepositToDelete(FormLeaseDeposit.createFromModel(deposit));
+      setDeleteModalWarning(true);
+    }
+  };
+
+  const onReturnDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditReturnValue(FormLeaseDepositReturn.createEmpty(deposit));
+      setShowReturnEditModal(true);
+    }
+  };
+
+  const onDeleteDepositConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositToDelete) {
+      const updatedLease = await removeLeaseDeposit({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositToDelete(undefined);
+        setDeleteModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  const onDeleteDepositReturnConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositReturnToDelete) {
+      const updatedLease = await removeLeaseDepositReturn({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDepositReturn.toInterfaceModel(depositReturnToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositReturnToDelete(undefined);
+        setDeleteReturnModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  /**
+   * Send the save request (either an update or an add). Use the response to update the parent lease.
+   * @param depositForm
+   */
+  const onSaveDeposit = async (depositForm: FormLeaseDeposit) => {
+    console.log('saving deposit!', depositForm);
+    if (lease && lease.id && lease.rowVersion) {
+      let request: IParentConcurrencyGuard<ILeaseSecurityDeposit> = {
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositForm),
+      };
+      const updatedLease = await updateLeaseDeposit(request);
+      if (!!updatedLease?.id) {
+        setLease(updatedLease);
+        setEditDepositValue(FormLeaseDeposit.createEmpty());
+        setShowEditModal(false);
+      }
+    } else {
+      console.log('Lease information incomplete');
```

Comment:
we can leave this in if you change it to a console.error

Reply:
I think I'd be tempted to throw an exception in cases like this, otherwise the user will interact with the page and no visible changes will occur.

Reply:
I think ideally we would have a toast for things like this since